barman/cloud_providers/aws_s3.py

```diff
@@ -774,13 +774,38 @@ def take_snapshot_backup(self, backup_info, instance_identifier, volumes):
             account_id=snapshot_resp["OwnerId"],
         )
 
+    def _delete_snapshot(self, snapshot_id):
+        """
+        Delete the specified snapshot.
+
+        :param str snapshot_id: The ID of the snapshot to be deleted.
+        """
+        try:
+            self.ec2_client.delete_snapshot(SnapshotId=snapshot_id)
+        except ClientError as exc:
+            error_code = exc.response["Error"]["Code"]
+            # If the snapshot could not be found then deletion is considered successful
```

Comment:
Tbh, if not found, we should add a log line at least at debug level, if not info, where we say "snapshot not found/doesn't exists", keeping track of events like this one helps debugging messy situations